server/ControlPlane/Versioning/README.Versioning.md

```diff
@@ -0,0 +1,154 @@
+# Tyger Server API Versioning
+
+## Server
+
+The Tyger server uses the [Asp.Versioning.Http library for ASP.NET Core Minimal APIs](https://github.com/dotnet/aspnet-api-versioning/wiki), which requires us to declare a set of server-supported API versions, then specify which versions apply to each endpoint.
+
+The API versions offered by the Tyger server are declared in `Versioning.cs`.
+The Tyger server's top-level route (`/`) is configured as a versioned API, so, by default, all endpoints in this route will support all offered API versions.
+Each endpoint can then be mapped to specific API versions as needed. See below for examples.
+
+## Client
+
+Clients must request a specific API version via the query parameter `api-version` for all endpoints except those marked as version neutral (e.g. `/metadata`).
+
+The Tyger client will automatically use the most recent API version it knows, but this can be overridden by the (*undocumented*) environment variable `TYGER_CLIENT_API_VERSION`, e.g.
```

Comment:
Were you thinking that commands could inpect this and downgrade their payloads or were you thinking that this would simply change the query parameter. In the latter case, it could be a dangerous thing to do (fields/parameters could have new meanings, etc.)

Reply:
Agreed. It was useful to toggle the query parameter in testing, but in practice, not a good idea. In the future, yes, commands could adapt to different API versions.